### ISSUE TITLE
False Positive: method-hidden with @functools.cached_property

### ISSUE THREAD

**Issue by Gidgidonihah** [pylint-dev/pylint#8753]
### Bug description

When decorating a method with `@functools.cached_property` pylint produces the error `method-hidden` whereas decorating it with `@property` does not.

example.py:
```python
# pylint: disable=too-few-public-methods
"""A bug report example."""
import functools


class Base:
    """Generic base class."""

    def __init__(self, example):
        self.example = example


class Extended(Base):
    """Class showing the method-hidden error."""

    @functools.cached_property
    def example(self):
        """
        This is a property and should be cached, but should not product method-hidden.
        """
        return "expensive function"
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
************* Module example
example.py:17:4: E0202: An attribute defined in example line 10 hides this method (method-hidden)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 0.00/10, +2.86)
```

### Expected behavior

The `method-hidden` error should not be emitted.

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.10.6 (main, Aug 31 2022, 16:28:02) [Clang 13.1.6 (clang-1316.0.21.2.5)]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. It still affects pylint 3.0.0b1, I think it's fair to also exclude cached_property from this check.
